provisioner/provision.js

```diff
@@ -79,61 +80,90 @@ var stats = {
   failure: 0,
 };
 
+// For when you want to be really certain that the program will
+// exit
+function exitTimer(time) {
+  var t = time || 30000;
+  setTimeout(() => {
+    debug('hey, so you probably are trying to figure out');
+    debug('why this process suddenly disappeared.  a major');
+    debug('error occured and you only get %d ms to exit after', t);
+    debug('before we process.exit(44). handle things faster!');
+    process.exit(44);
+  }, t);
+}
+
 /**
  * Start running a provisioner.
  */
 Provisioner.prototype.run = function () {
-  var that = this;
-
   this.__keepRunning = true;
+
   this.__watchDog.on('expired', function () {
-    debug('[alert-operator] Provisioner is stuck, killing');
+    debug('[alert-operator] provisioning iteration exceeded max time');
+    exitTimer(MAX_KILL_TIME);
     throw new Error('WatchDog expired');
   });
+
   this.__watchDog.start();
 
-  function provisionIteration () {
+  var provisionIteration = async () => {
```

Comment:
Okay, I love that you're starting to use `async` one of the really cool things is that you no longer need to make an infinite loop, your code boils down to this:
```
do {
  debug('starting iteration %d, successes %d, failures %d',
          this.iterations,
          this.success,
          thís.failures);
  this.__watchDog.touch();

  // We should make sure that we're not just permanently failing
  // We also don't want to
  if (this.failures > MAX_FAILURES) {
    debug('[alert-operator] dieing after %d failures', MAX_FAILURES);
    exitTimer(MAX_KILL_TIME);
    throw new Error('provisioner is failing a lot');
  }

  this.iterations++;
  try {
    await this.runAllProvisionersOnce();
    stats.success++;
  } catch (err) {
    stats.failures++;
    debug('[alert-operator] provisioning iteration failure');
    if (err.stack) {
      debug('[alert-operator] stack: %s', err.stack.replace('\n', '\\n'));
    }
  }

  await base.testing.sleep(this.provisionIterationInterval);
} while(that.__keepRunning && !process.env.PROVISION_ONCE);
```